Build failure: New manual mscorlib shim

@dotnet-mc-bot commented on [Thu Jun 08 2017](https://github.com/dotnet/core-eng/issues/899)

Build : Master - 20170608.02 (Product Build)
Failing configurations:
- Windows
  - AllConfigurations-Release-x64
[Mission Control Build Info](https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/build~2Fproduct~2F/build/20170608.02/workItem/Orchestration/analysis/external/Visual%20Studio%20Build%20Information)

https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=792955 

```

Failed due to a product build issue:
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(13,49): error CS0246: The type or namespace name 'Attribute' could not be found (are you missing a using directive or an assembly reference?) [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(12,6): error CS0518: Predefined type 'System.Object' is not defined or imported [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(12,6): error CS0246: The type or namespace name 'AttributeUsageAttribute' could not be found (are you missing a using directive or an assembly reference?) [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(12,6): error CS0246: The type or namespace name 'AttributeUsage' could not be found (are you missing a using directive or an assembly reference?) [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(12,21): error CS0518: Predefined type 'System.Object' is not defined or imported [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]
2017-06-08T12:14:35.7223367Z E:\A\_work\352\s\corefx\Tools\BlockReflectionAttribute.cs(12,21): error CS0103: The name 'AttributeTargets' does not exist in the current context [E:\A\_work\352\s\corefx\src\shims\manual\mscorlib.csproj]

```

Looks related to @weshaggard’s change here: https://github.com/dotnet/corefx/commit/ca49bfbfd64351fddba56f2141b90841b8561eb7 
But is confusing since it seems to be specific to the release build.

---

@weshaggard commented on [Thu Jun 08 2017](https://github.com/dotnet/core-eng/issues/899#issuecomment-307175737)

It is a release only issue which is why it got through CI. CI only runs debug version of AllConfigurations build leg.

--- a/dir.targets
+++ b/dir.targets
@@ -26,7 +26,7 @@
 
   <ItemGroup Condition="'@(BinPlaceConfiguration)' == ''">
     <!-- binplace to directories for the target vertical -->
-    <BinPlaceConfiguration Include="$(BuildConfiguration)">
+    <BinPlaceConfiguration Include="$(_bc_TargetGroup)-$(_bc_OSGroup)">
       <RefPath>$(BuildConfigurationRefPath)</RefPath>
       <RuntimePath>$(RuntimePath)</RuntimePath>
     </BinPlaceConfiguration>

